pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -130,32 +131,34 @@ func (e *OpenTsdbExecutor) parseResponse(query OpenTsdbQuery, res *http.Response
 		return nil, fmt.Errorf("request failed, status: %s", res.Status)
 	}
 
-	var data []OpenTsdbResponse
-	err = json.Unmarshal(body, &data)
+	var responseData []OpenTsdbResponse
+	err = json.Unmarshal(body, &responseData)
 	if err != nil {
 		plog.Info("Failed to unmarshal opentsdb response", "error", err, "status", res.Status, "body", string(body))
 		return nil, err
 	}
 
-	for _, val := range data {
-		series := plugins.DataTimeSeries{
-			Name: val.Metric,
-		}
+	frames := data.Frames{}
+	for _, val := range responseData {
+		timeVector := make([]time.Time, 0, len(val.DataPoints))
+		values := make([]float64, 0, len(val.DataPoints))
+		name := val.Metric
 
 		for timeString, value := range val.DataPoints {
-			timestamp, err := strconv.ParseFloat(timeString, 64)
+			// we need to change here
+			timestamp, err := strconv.ParseInt(timeString, 10, 64)
 			if err != nil {
 				plog.Info("Failed to unmarshal opentsdb timestamp", "timestamp", timeString)
 				return nil, err
 			}
-			series.Points = append(series.Points, plugins.DataTimePoint{
-				null.FloatFrom(value), null.FloatFrom(timestamp),
-			})
+			timeVector = append(timeVector, time.Unix(timestamp, 0).UTC())
+			values = append(values, value)
 		}
-
-		queryRes.Series = append(queryRes.Series, series)
+		frames = append(frames, data.NewFrame(name,
+			data.NewField("time", nil, timeVector),
+			data.NewField("value", nil, values)))
```

Comment:
Generally OpenTSDB is a label based system, so the second argument of `NewField` should likely be `data.Labels{the labels for the series}`. Without this, the new alerting system won't be able to to multi-dimensional alerts and it will not work well Server Side Expressions (SSE).

Reply:
thanks Kyle, I am not sure though what the labels should be

Reply:
maybe the labels could be solved on an different PR, since it hasn't been supported so far

Reply:
That sounds reasonable to me. In addition I cannot find any issue for OpenTSDB, tags and alerting why it doesn't seem like a lot of users uses OpenTSDB any longer. Lets wait with this until requested